pkg/discovery/lifecycle.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package discovery
+
+import (
+	"context"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+// GracefulDisconnect is called when a node alerts the network they're
+// going offline for a short period of time with intent to come back
+func (d *Discovery) GracefulDisconnect(id storj.NodeID) {
+}
+
+// ConnFailure implements the Transport Observer interface `ConnFailure` function
+func (d *Discovery) ConnFailure(ctx context.Context, node *pb.Node, err error) {
```

Comment:
You could probably still call these ConnectionFailure and ConnectionSuccess since they're in a different package

Reply:
Well If we can change the observer interface names, but just a suggestion not vital

Reply:
Oh yeah, I agree with changing Conn -> Connection across the board